include/LogicTerm/TermImpl.hpp

```diff
@@ -23,76 +23,76 @@ namespace logicbase {
-            lb(lb), id(getNextId(lb)), name(std::to_string(id)), cType(cType) {}
+        explicit TermImpl(CType type, Logic* logic = nullptr):
+            lb(logic), id(getNextId(lb)), name(std::to_string(id)), cType(type) {}
 
-        explicit TermImpl(std::string name, CType cType, Logic* lb = nullptr,
-                          int16_t bvSize = 0):
-            lb(lb),
-            id(getNextId(lb)), name(std::move(name)), bvSize(bvSize), cType(cType) {}
+        explicit TermImpl(std::string n, CType type, Logic* logic = nullptr,
+                          int16_t bvs = 0):
+            lb(logic),
+            id(getNextId(lb)), name(std::move(n)), bvSize(bvs), cType(type) {}
 
-        explicit TermImpl(std::string name, int64_t id, CType cType,
-                          Logic* lb = nullptr):
-            lb(lb),
-            id(id), name(std::move(name)), cType(cType) {}
+        explicit TermImpl(std::string n, const uint64_t identifier, CType type,
+                          Logic* logic = nullptr):
+            lb(logic),
+            id(identifier), name(std::move(n)), cType(type) {}
 
         explicit TermImpl(OpType op, const std::initializer_list<LogicTerm>& n,
-                          CType cType = CType::BOOL, Logic* lb = nullptr);
+                          CType type = CType::BOOL, Logic* logic = nullptr);
 
         explicit TermImpl(OpType op, const std::vector<LogicTerm>& n,
-                          CType cType = CType::BOOL, Logic* lb = nullptr);
+                          CType type = CType::BOOL, Logic* logic = nullptr);
 
-        explicit TermImpl(OpType op, const LogicTerm& a, CType cType = CType::BOOL,
-                          Logic* lb = nullptr);
+        explicit TermImpl(OpType op, const LogicTerm& a, CType type = CType::BOOL,
+                          Logic* logic = nullptr);
         explicit TermImpl(OpType op, const LogicTerm& a, const LogicTerm& b,
-                          CType cType = CType::BOOL, Logic* lb = nullptr);
+                          CType type = CType::BOOL, Logic* logic = nullptr);
         explicit TermImpl(OpType op, const LogicTerm& a, const LogicTerm& b,
-                          const LogicTerm& c, CType cType = CType::BOOL,
-                          Logic* lb = nullptr);
+                          const LogicTerm& c, CType type = CType::BOOL,
+                          Logic* logic = nullptr);
         TermImpl(const TermImpl& other);
         explicit TermImpl(const LogicTerm& other);
 
-        static int64_t getNextId(Logic* lb = nullptr) {
-            if (lb == nullptr) {
+        static uint64_t getNextId(Logic* logic = nullptr) {
+            if (logic == nullptr) {
                 return gid++;
             }
-            {
-                return lb->getNextId();
-            }
+            return logic->getNextId();
         }
-        [[nodiscard]] static std::string getStrRep(OpType opType);
+        [[nodiscard]] static std::string getStrRep(OpType type);

```

Comment:
Inconsistent change of parameter, probably should be op?
```suggestion
        [[nodiscard]] static std::string getStrRep(OpType op);

```

Reply:
Yeah 👍🏻